docs/testing/asserts.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Asserts
+type: docs
+menu:
+  docs:
+    parent: 'Testing'
+weight: 4
+---
+
+# Asserts
+
+Test asserts are the part of a [test step](/docs/testing/steps) that define the state to wait for Kubernetes to reach. It is possible to match specific objects by name as well as match any object that matches a defined state.
+
+* [Format](#format)
+* [Getting a resource from the cluster](#getting-a-resource-from-the-cluster)
+* [Listing resources in the cluster](#listing-resources-in-the-cluster)
+
+# Format
+
+The test assert file for a test step is found at `$index-assert.yaml`. So, if the test step index is `00`, the assert should be called `00-assert.yaml`. This file can contain any number of objects to match on. If the objects have a namespace set, it will be respected, but if a namespace is not set, then the test harness will look for the objects in the test case's namespace.
```

Comment:
maybe note somewhere that the assert inside step is optional? but it then mean that the next step is run immediately after, which might not be what everyone wants

Reply:
Good call, we've got a couple of tests written this way just for clarity's sake.